x-pack/plugins/uptime/public/components/common/monitor_tags.tsx

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useState } from 'react';
+import { EuiBadge, EuiBadgeGroup } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { Ping } from '../../../common/runtime_types/ping';
+import { MonitorSummary } from '../../../common/runtime_types/monitor';
+import { useFilterUpdate } from '../../hooks/use_filter_update';
+import { useGetUrlParams } from '../../hooks';
+import { parseCurrentFilters } from '../overview/monitor_list/columns/monitor_name_col';
+import { EXPAND_TAGS_LABEL } from '../overview/monitor_list/columns/translations';
+
+interface Props {
+  ping?: Ping | null;
+  summary?: MonitorSummary;
+}
+
+const getTagsFromSummary = (summary: MonitorSummary) => {
+  let tags = new Set<string>();
+  summary.state.summaryPings.forEach((ping) => {
+    tags = new Set([...tags, ...(ping?.tags ?? [])]);
+  });
+
+  return [...tags];
+};
+
+const getTagsFromPing = (ping: Ping) => {
+  return ping?.tags ?? [];
+};
+
+const getFilterLabel = (tag: string) => {
+  return i18n.translate('xpack.uptime.monitorList.tags.filter', {
+    defaultMessage: 'Filter all monitors with tag {tag}',
+    values: {
+      tag,
+    },
+  });
+};
+
+export const MonitorTags = ({ ping, summary }: Props) => {
+  const [toDisplay, setToDisplay] = useState(5);
+  let tags: string[];
+
+  if (summary) {
+    // summary in case of monitor list
+    tags = getTagsFromSummary(summary!);
+  } else {
+    tags = getTagsFromPing(ping!);
+  }
+
+  const tagsToDisplay = tags.slice(0, toDisplay);
+
+  const { absoluteDateRangeStart, absoluteDateRangeEnd, ...params } = useGetUrlParams();
+
+  const currFilters = parseCurrentFilters(params.filters);
+
+  const [filterType, setFilterType] = useState<string[]>(currFilters.get('tags') ?? []);
+
+  useFilterUpdate('tags', filterType);
+
+  if (tags.length === 0) {
+    return null;
+  }
+
+  return (
+    <EuiBadgeGroup>
+      {tagsToDisplay.map((tag) =>
+        // filtering only makes sense in monitor list, where we have summary
```

Comment:
On the details page this should link back to the monitor overview page / monitor list.

Imagine you're a user viewing a monitor and want to jump to all monitors that share a similar category

Reply:
@andrewvc make sense, i have pushed the update